lib/galaxy/util/xml_macros.py

```diff
@@ -17,13 +17,30 @@ def load_with_references(path):
 
     macro_paths = _import_macros(root, path)
 
+    # temporarily remove the children of the macros node
+    # and create a copy. this is done because this allows
+    # to iterate over all expand nodes of the tree
+    # that are not included in the macros node
+    macros_el = _macros_el(root)
+    if macros_el is not None:
+        macros_copy = deepcopy(macros_el)
```

Comment:
deepcopy essentially pickles and unpickles the object, this could be detrimental to startup speed.

Reply:
True. I remember that if I do not deepcopy then the following clear also removes the copied elements. Maybe calling `remove(subelement)` for each child helps. 

In the end this is only a workaround: I was not able to find a way to iterate only the `expand` elements that are not included in the `macros` subtree. With lxml it might be possible, but it appears to me that the code needs to be compatible with xml from the stdlib and lxml, or am I wrong?

Reply:
You could import `LXML_AVAILABLE` and make that conditional on this (if it works).